bundles/org.openhab.binding.kostalinverter/src/main/java/org/openhab/binding/internal/kostal/inverter/secondgeneration/SecondGenerationHandler.java

```diff
@@ -0,0 +1,359 @@
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounterExt++;
+            }
+        }
+
+        // Create and update actual values for each channelPostExtExt
+        int channelValuesCounterExtExt = 0;
+        for (SecondGenerationChannelConfiguration cConfig : channelConfigsExtExt) {
+            Channel channel = getThing().getChannel(cConfig.id);
+            State state = getState(channelPostsExtExt[channelValuesCounterExtExt], cConfig.unit);
+
+            // Update the channel
+            if (state != null) {
+                updateState(channel.getUID().getId(), state);
+                channelValuesCounterExtExt++;
+            }
+        }
+    }
+
+    // Helper method to handleCommand method
+    public final void preSetExecuteConfigurationChanges(HttpClient httpClient, String url, String username,
+            String password, String dxsEntriesConf, String valueConfiguration) {
+        try {
+            SecondGenerationConfigurationHandler.executeConfigurationChanges(httpClient, url, username, password,
+                    dxsEntriesConf, valueConfiguration);
+        } catch (Exception handleCommandException) {
+            logger.debug("Handle command for {} on channel {}: {}: {}: {}: {}: {}", thing.getUID(), httpClient, url,
+                    username, password, dxsEntriesConf, valueConfiguration);
+        }
+    }
+
+    @SuppressWarnings("null")
+    public static String callURL(String myURL) {
+        StringBuilder sb = new StringBuilder();
+        URLConnection urlConn = null;
+        InputStreamReader in = null;
+        try {
+            URL url = new URL(myURL);
+            urlConn = url.openConnection();
+            if (urlConn != null) {
+                urlConn.setReadTimeout(60 * 1000);
+            }
+            if (urlConn != null && urlConn.getInputStream() != null) {
+                try {
+                    in = new InputStreamReader(urlConn.getInputStream(), Charset.defaultCharset());
+                    BufferedReader bufferedReader = new BufferedReader(in);
+                    if (bufferedReader != null) {
+                        int cp;
+                        while ((cp = bufferedReader.read()) != -1) {
+                            sb.append((char) cp);
+                        }
+                        bufferedReader.close();
+                    }
+                } catch (IOException getInputStreamReaderException) {
+                    throw new RuntimeException("Exception while calling URL:" + myURL, getInputStreamReaderException);
+                }
+            }
+
+            in.close();
+        } catch (IOException callURLException) {
+            throw new RuntimeException("Exception while calling URL:" + myURL, callURLException);
+
+        }
+        return sb.toString();
+    }
+
+    @SuppressWarnings({ "null", "unused" })
+    private State getState(String value, Unit<?> unit) {
```

Comment:
You could annotated `unit` with `@Nullable`. Then, the suppression can be removed.
```suggestion
    private State getState(String value, @Nullable Unit<?> unit) {
```